src/posit/connect/env.py

```diff
@@ -0,0 +1,220 @@
+from __future__ import annotations
+
+from typing import (
+    Any,
+    Dict,
+    List,
+    Iterable,
+    MutableMapping,
+    Optional,
+    overload,
+)
+
+from requests import Session
+
+from . import urls
+from .config import Config
+from .resources import Resources
+
+
+class EnvVars(Resources):
+    def __init__(
+        self, config: Config, session: Session, content_guid: str
+    ) -> None:
+        super().__init__(config, session)
+        self.content_guid = content_guid
+
+    def __setitem__(self, key: str, value: str, /) -> None:
+        """Set environment variable.
+
+        Set the environment variable for content.
+
+        Parameters
+        ----------
+        key : str
+            The name of the environment variable to set.
+        value : str
+            The value assigned to the environment variable.
+
+        Examples
+        --------
+        >>> vars = EnvVars(config, session, content_guid)
+        >>> vars["DATABASE_URL"] = (
+        ...     "postgres://user:password@localhost:5432/database"
+        ... )
+        """
+        self.update({key: value})
+
+    def __delitem__(self, key: str, /) -> None:
+        """Delete the environment variable.
+
+        Parameters
+        ----------
+        key : str
+            The name of the environment variable to delete.
+
+        Examples
+        --------
+        >>> vars = EnvVars(config, session, content_guid)
+        >>> del vars["DATABASE_URL"]
+        """
+        self.update({key: None})
```

Comment:
These methods are included to provide an idiomatic interface for interfacing with environments. Unfortunately, the __getitem__ dunder method cannot be implemented since the environments API doesn't return values.

Reply:
Ah cool so you do have these!

I'd add `keys()` as an alias to `find()`, as that seems like the more natural way to list the names of the env vars as dict.

`__getitem__` should probably exist but error with a message explaining that the values are encrypted so we can't show them. Likewise `items()` and similar.